tools/roxvet/analyzers/uncheckedifassign/analyzer.go

```diff
@@ -60,31 +59,21 @@ func checkAnyObjectUsedIn(expr ast.Expr, objs map[*ast.Object]struct{}) bool {
 }
 
 func run(pass *analysis.Pass) (interface{}, error) {
-	nodeFilter := []ast.Node{(*ast.File)(nil), (*ast.IfStmt)(nil)}
+	nodeFilter := []ast.Node{
+		(*ast.IfStmt)(nil),
+	}
 
 	inspectResult := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
-	inspectResult.Nodes(nodeFilter, func(n ast.Node, push bool) bool {
-		if !push {
-			return false
-		}
-
-		if file, _ := n.(*ast.File); file != nil {
-			if len(file.Comments) > 0 && len(file.Comments[0].List) > 0 {
-				return !generatedFileCommentRegex.MatchString(file.Comments[0].List[0].Text)
-			}
-		}
-
-		ifStmt, _ := n.(*ast.IfStmt)
-		if ifStmt == nil || ifStmt.Init == nil {
-			return false
+	common.FilteredPreorder(inspectResult, common.Not(common.IsGeneratedFile), nodeFilter, func(n ast.Node) {
```

Comment:
why switch this to preorder?

Reply:
The only reason to use `Nodes` here was to allow preventing the traversal of generated `*ast.File` nodes. Doing so in a preorder traversal is possible, but much less intuitively so than in a DFS traversal. However, given that we now have a generic utility method, there's no harm in implementing the file skipping logic once and then just using it here. And ultimately, if we can use `Preorder`, we should, because it's simpler (as is apparent from the callback function signature).